FoodOrderingSystem.py

```diff
@@ -0,0 +1,700 @@
+        """ Setup engine, return engine if exist """
+        if self.engine:
+            return
+        self.engine = create_engine(db_creation, echo=False, pool_recycle=3600)
+        self.Session.configure(bind=self.engine)
+
+    def bootstrap(self):
+        """ Connects to engine and creates tables """
+        connection  = None
+
+        for i in range(2):
+            try:
+                connection = self.engine.connect()
+            except:
+                print("DB Server is not up yet, Retrying..")
+                time.sleep(i * 5)
+                continue
+        if not connection:
+            raise Exception("Couldn't connect to DB Server even after retries!")
+
+        Base.metadata.create_all(self.engine)
+        connection.close()
+
+class Employee:
+    """ Restaurant's end operation """
+    # View order and status by customer id
+    # View all orders/status
+    # Delete and update food category and food details and delivery person
+    # View checkout orders/en route orders/ delieverd orders
+    # View revenue of a particular date/month
+
+    def add_food_category(self, category_name, session):
+        """ Insert a new food category """
+        category = FoodCategory(name=category_name)
+        session.add(category)
+        try:
+            session.commit()
+            return category
+        except IntegrityError:
+            session.rollback()
+            raise Exception("Category exist")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def add_food_details(self, category_id, food_name, price, session):
+        """ Insert new food details """
+        details = FoodDetails(category_id=category_id, food_name=food_name, price=price)
+        session.add(details)
+        try:
+            session.commit()
+            return details
+        except IntegrityError:
+           session.rollback()
+           raise Exception("Food exist")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def add_delivery_person(self, delivery_person_name, delivery_person_phone, session):
+        """ Insert new delivery person """
+        details = DeliveryPerson(delivery_person_name=delivery_person_name, delivery_person_phone=delivery_person_phone)
+        session.add(details)
+        try:
+            session.commit()
+            return details
+        except IntegrityError:
+           session.rollback()
+           raise Exception("Delivery person exist")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def assign_deliver_person_to_deliver_order(self, order_id, delivery_person_id, session):
+        """ Update CustOrderStatus table to add deliver person to deliver order """
+        update = session.query(CustOrderStatus).filter(CustOrderStatus.order_id == order_id).update({CustOrderStatus.delivery_person_id: delivery_person_id}, synchronize_session=False)
```

Comment:
What do you understand by `synchronize_session=False`? why you used it?

Reply:
synchronize_session updates the attributes on objects in the session. There are three strategies (False, fetch and evaluate).

False is used once a session has been expired so this strategies is more reliable to update the attributes. 

https://docs.sqlalchemy.org/en/13/orm/query.html